bin.js

```diff
@@ -45,13 +50,21 @@ function getConfig(file) {
     // point to the particular config
     configFile: file,
   })
-  var config = cliEngine.getConfigForFile(file)
-  return config
+  return cliEngine.getConfigForFile(file)
+}
+
+function getCurrentRules(conf) {
+  var rules = Object.keys(conf.rules)
+  return rules
 }
 
-function getRules() {
-  var configFile = getConfigFile()
-  var config = getConfig(configFile)
-  var rules = Object.keys(config.rules)
+function getPluginRules(conf) {
+  var rules = []
+  var plugins = conf.plugins
+  if (plugins) {
+    plugins.forEach(function normalizePluginRule(plugin) {
+      rules.concat(require('eslint-plugin-' + plugin).rules)
```

Comment:
There exist some issues with this line of code:

1.) You have to reassign the result of `rules.concat` back to `rules`, because [`concat` does not alter the Array it's called from, but returning a new instance of Array](https://developer.mozilla.org/de/docs/Web/JavaScript/Reference/Global_Objects/Array/concat).
```
new_array = old_array.concat(value1[, value2[, ...[, valueN]]])
```

2.) `require('eslint-plugin-' + plugin).rules` will return an Object or `undefined`, but not an Array.

The following should suffice:
```JavaScript
var pluginRules = require('eslint-plugin-' + plugin).rules;
if (typeof pluginRules === 'object') {
  rules = rules.concat(Object.keys(pluginRules));
}
```